dungeon/src/task/quizquestion/Quiz.java

```diff
@@ -0,0 +1,152 @@
+ * the type of question.
+ *
+ * <p>A {@link Quiz} can be a Single-Choice, a Multiple-Choice, or a Free-text question. The type is
+ * stored as a {@link QuizType}. If the question is asked via the UI, the {@link QuizUI} will
+ * configure the UI for the question based on that type. The type can be accessed via {@link
+ * #type()}.
+ *
+ * <p>Add a {@link QuizContent} answer by using the {@link #addAnswer(QuizContentType, String)}
+ * method. Use the {@link #contentStream()} method to get the answers as a stream.
+ *
+ * <p>The question will be stored as {@link QuizContent} and can be accessed via {@link
+ * #question()}.
+ */
+public class Quiz extends Task {
+
+    private final QuizType type;
+    private final QuizContent question;
+
+    /**
+     * Create a new {@link Quiz} with the given configuration.
+     *
+     * <p>This will create a new {@link QuizContent} instance as the question reference.
+     *
+     * <p>The {@link Quiz} will not have any answers, use {@link #addAnswer(QuizContentType,
+     * String)} to add possible answers to the question.
+     *
+     * @param type Type of the question (e.g., single-choice)
+     * @param questionContentType What does the question contain? Just text, just a path to an
+     *     image, or both text and a path to an image?
+     * @param questionText The question itself (can contain a path to images).
+     */
+    public Quiz(final QuizType type, QuizContentType questionContentType, String questionText) {
+        super();
+        this.type = type;
+        taskText(questionText);
+        question = new QuizContent(this, questionContentType, questionText);
+    }
+
+    /**
+     * Get the type of the question.
+     *
+     * @return The type of the question.
+     */
+    public QuizType type() {
+        return type;
+    }
+
+    /**
+     * Create a {@link QuizContent} answer instance with the given configuration as a possible
+     * answer for this question.
+     *
+     * @param type What does the answer contain? Just text, just a path to an image, or both text
+     *     and a path to an image?
+     * @param content The answer itself (can contain a path to images).
+     */
+    public void addAnswer(QuizContentType type, String content) {
+        addContent(new QuizContent(this, type, content));
+    }
+
+    /**
+     * Get the question instance.
+     *
+     * @return The question instance.
+     */
+    public QuizContent question() {
+        return question;
+    }
+
+    /**
+     * The QuizQuestionContentType enum represents the different types of content that can be
+     * associated with a quiz question. The available types are TEXT, IMAGE, and TEXT_AND_IMAGE.
+     * TEXT represents a question or answer choice as text. IMAGE represents a question or answer
+     * choice as an image. TEXT_AND_IMAGE represents a question or answer choice as both text and an
+     * image.
+     */
+    public enum QuizContentType {
```

Comment:
sollte dieses enum eher teil von quizcontent sein?

Reply:
Eigentlich ja, aber ich fand `Quiz.QuizContent.QuizContentType.$VALUE` echt bescheiden.....

Reply:
einerseits bin ich da bei dir, andererseits gehört das enum einfach hier nicht hin. vermutlich müssen wir doch nochmal über die modellierung nachdenken.

Reply:
Vielleicht ist dieser ganze inner-class kram auch ne Schnaps Idee gewesen. 
Ich fands irgendwie ganz sexy, da ich so alles beisamen halten konnte.

Man könnte auch einfach alles in je eine Klasse und in das packe `quiz` werfen. Dann verlier ich zwar meinen sexy `private` Konstruktor von `QuizContent`, aber naja.... Irgendeinen tot muss ich sterben

Reply:
Ich habe das enum in die richtige inner class geschoben, die klassen aber umbenannt. 
Da ich eh `Quiz.QuizContent.QuizContentType` aufrufen muss, kann ich daraus aus `Quiz.Content.Type` machen, aus der namensstruktur wird ja klar, dass es sich bei `Content` um Content für ein `Quiz` handeln muss.